### PR TITLE
bpo-39487: Merge duplicated _Py_IDENTIFIER identifiers in C code

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue39487

### CODE REVIEW

**Review by brettcannon — APPROVED** [review]

File: Python/import.c [hunk 1]
```diff
@@ -1568,7 +1570,6 @@ resolve_name(PyThreadState *tstate, PyObject *name, PyObject *globals, int level
 {
     _Py_IDENTIFIER(__spec__);
```
Comment (brettcannon) [inline Python/import.c]:
```suggestion
```

**Review by brettcannon — CHANGES_REQUESTED** [review]

**Comment by ncoghlan** [comment]
We can't make this change, as it means the statics get initialised before the Python interpreter has been initialised, and won't be reinitialised if the interpreter is destroyed and recreated.

**Comment by ncoghlan** [comment]
My apologies, my comment above was based on an outdated understanding of how the identifier structs get initialised (it's the usage that initialises them, not the declaration).

That means this is a useful refactoring to help identify blockers to full subinterpreter support.

**Comment by shihai1991** [comment]
I have made the requested changes; please review again

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
Thanks, @shihai1991 !

**Pull request merged**